src/fractal-registry.ts

```diff
@@ -6,32 +6,33 @@ import {
 import { DAO } from '../generated/schema';
 
 const loadOrCreateDAO = (address: Bytes): DAO => {
-  let dao = DAO.load(address); // Using address as ID
-  if (!dao) {
-    dao = new DAO(address);
-    dao.address = address; // But also keep address field on DAO entity in case we would want to use something else as ID
-    dao.hierarchy = [];
+  const existingDao = DAO.load(address); // Using address as ID
+  if (existingDao) {
+    return existingDao;
   }
 
-  return dao;
+  const newDao = new DAO(address);
+  newDao.address = address; // But also keep address field on DAO entity in case we would want to use something else as ID
+  newDao.hierarchy = [];
+  return newDao;
 };
 
 export function handleFractalNameUpdated(event: FractalNameUpdatedEvent): void {
-  let dao = loadOrCreateDAO(event.params.daoAddress);
+  const dao = loadOrCreateDAO(event.params.daoAddress);
   dao.name = event.params.daoName;
-
   dao.save();
 }
 
 export function handleFractalSubDAODeclared(event: FractalSubDAODeclaredEvent): void {
-  let subDAO = loadOrCreateDAO(event.params.subDAOAddress);
+  const subDAO = loadOrCreateDAO(event.params.subDAOAddress);
+  if (subDAO.parentAddress !== null) {
+    return;
+  }
+
```

Comment:
Here's the good stuff. Early exit if the child already has a parent.

Another way of saying it, is that once a child/parent relationship is formed for the first time, it can't be changed.

CURRENTLY what happens (if you read the original code closely) is that anyone can declare an existing child as their own, which will update the child's "parent address" pointer to this new address, and then set the child address as a child of this new address. The original parent, however, will still see child in their own children hierarchy. From the child's point of view though, they only see the new parent as their parent.

Reply:
Ah, good catch. I guess my question is, should we completely make it unchangeable? is there a way to allow the current parentAddress to update still change it?

Reply:
I thought about that too. It might be good to allow a parent to "disown" a child DAO, which would remove it from their list of `hierarchy` addresses.

When we decide to build this functionality into the frontend, we can always modify this subgraph and redeploy/reindex.